onnxruntime/test/providers/kernel_def_hash_test.cc

```diff
@@ -155,6 +155,11 @@ TEST(KernelDefHashTest, DISABLED_PrintCpuKernelDefHashes) {
   std::cout << DumpKernelDefHashes(cpu_kernel_def_hashes) << "\n";
 }
 
+// TODO: Enable this test when optional type isdisabled when we have a way to conditionally
+// change the expected hash list (right now they are read from a file as shown below).
+// The actual hashes of some operators change when the optional type is disabled.
+#if !defined(DISABLE_OPTIONAL_TYPE)
```

Comment:
I don't think we can allow them to, so we might need a minimal DataTypeImpl for the 'optional' types that just has a working ToString for the kernel hashing. 

Maybe a generic 'disabled' DataTypeImpl class where everything but ToString is stubbed to minimize the cost would be possible.